lib/ruby_lsp/global_state.rb

```diff
@@ -166,8 +168,24 @@ def detect_typechecker(dependencies)
     end
 
     sig { returns(T::Boolean) }
-    def bin_rails_present
-      File.exist?(File.join(workspace_path, "bin/rails"))
+    def rails_app?
+      config = rails_app_config
+      return false unless config
+
+      result = Prism.parse(config)
+      return false unless result
+
+      dispatcher = Prism::Dispatcher.new
+      listener = Listeners::RailsApp.new(dispatcher)
+      dispatcher.dispatch(result.value)
+
+      listener.rails_app
+    end
+
+    sig { returns(T.nilable(String)) }
+    def rails_app_config
+      path = File.join(workspace_path, "config/application.rb")
```

Comment:
I'm wondering if it's possible that the file is not right there, but would be nested deeper? I'm not too familiar with the different way people configure their project in relation with the workspace. At the same time, it might be best to stay with the simple solution.

Reply:
(and we already require that the Rails app is at the root of a workspace, so that's not a concern).